src/main/java/codeu/controller/RegisterServlet.java

```diff
@@ -36,5 +44,25 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
 
     }
 
+    /**
+     * Set up state for handling registration-related requests.
+     * This method is only called when running a server, not when running in a test
+     */
+
+    @Override
+    public void init() throws ServletException{
+        super.init();
+        setUserStore(UserStore.getInstance());
```

Comment:
Can you inject the user store instead of doing this?  That is, make it an argument to the constructor, and make the private member a final (private final UserStore)?  That way, you will never have a scenario in which you can have a RegisterServlet that has a null userStore; if you have successfully constructed a RegisterServlet, it will have all of its dependencies.  (This is called "dependency injection".  There are fancy frameworks for this, like Spring or Guice, but you don't have to use those to still get the benefit.)

Reply:
This makes sense. I think I fixed it.

Reply:
Actually, I didn't quite fix it, but I'm working on it